src/index.ts

```diff
@@ -90,44 +93,52 @@ class SpressoSDK {
     }
 
     async getPrice(request: PricingRequest, userAgent: string | undefined): Promise<PricingResponse> {
-        const response = await this.makeRequest(
-            'get',
-            request,
-            undefined,
-            userAgent
-        ).catch((err) => {
-            this.handleAxiosError(err);
-            return this.emptyResponse(request);
-        });
+        try {
+            const response = await this.makeRequest(
+                'get',
+                request,
+                undefined,
+                userAgent
+            ).catch((err) => {
+                this.handleAxiosError(err);
+                return this.emptyResponse(request);
+            });
 
-        if (response == null) {
+            if (response == null) {
+                return this.emptyResponse(request);
+            }
+
+            return response as PricingResponse;
+        } catch (err) {
             return this.emptyResponse(request);
         }
-
-        return response as PricingResponse;
     }
 
     async getPrices(requests: PricingRequest[], userAgent: string | undefined): Promise<PricingResponse[]> {
-        const response = await this.makeRequest(
-            'post',
-            undefined,
-            { requests },
-            userAgent
-        ).catch((err) => {
-            this.handleAxiosError(err);
-            return this.emptyResponses(requests);
-        });
+        try {
+            const response = await this.makeRequest(
+                'post',
+                undefined,
+                { requests },
+                userAgent
+            ).catch((err) => {
+                this.handleAxiosError(err);
+                return this.emptyResponses(requests);
+            });
 
-        if (response == null) {
+            if (response == null) {
+                return this.emptyResponses(requests);
+            }
+
+            return response as PricingResponse[];
+        } catch (err) {
             return this.emptyResponses(requests);
         }
-
-        return response as PricingResponse[];
     }
 
     private async authenticate(): Promise<void> {
         const now = new Date().getTime();
-        if (this.authToken != null && this.tokenExpiration != null && now < this.tokenExpiration) {
+        if (this.authToken != null && this.tokenExpiration != null && now < (this.tokenExpiration - AUTH_EXPIRATION_PAD_MS)) {
```

Comment:
ah so we want to reauthenticate if there is less than 30mins left for the token expiration, instead of waiting until it actually expires

Reply:
Exactly, because of timing issues like this (see timestamp of the log message on left and error on right)

<img width="1256" alt="Screen Shot 2023-11-30 at 23 12 36" src="https://github.com/spressoinsights/spresso-sdk-node/assets/33885037/701467c6-c6d5-48fc-82e0-d7c4ef47ff9b">